### PR TITLE
Improve path checking before route registration

### PR DESCRIPTION
**Pull request by peterver**
## Checklist

- [x] I have ensured my pull request is not behind the main or master branch of the original repository.
- [x] I have rebased all commits where necessary so that reviewing this pull request can be done without having to merge it first.
- [x] I have written a commit message that passes commitlint linting.
- [x] I have ensured that my code changes pass linting tests.
- [x] I have ensured that my code changes pass unit tests.
- [x] I have described my pull request and the reasons for code changes along with context if necessary.

fixes: #147, #148

### CODE REVIEW

**Comment by peterver** [comment]
@wengeezhang @etroynov kindly review once you find some time in your busy schedules

**Comment by etroynov** [comment]
@3imed-jaberi @niftylettuce can you review it?

File: test/lib/router.js [hunk 1]
```diff
@@ -993,7 +993,24 @@ describe('Router', function () {
           done();
         });
     });
+    
+    it ('correctly returns an error when not passed a path for verb-specific registration (gh-147)', function () {
+      const router = new Router();
+      try {
+        router.get(function (ctx, next) {});
```
Comment (etroynov) [inline test/lib/router.js:1000]: this arguments is not necessary `(ctx, next)`

**Review by etroynov — CHANGES_REQUESTED** [review]

**Comment by peterver** [comment]
> I also recommend you change the title, because someone may not understand that PR is about a specific issue.
> 
> It could be something like: `Improve path checking before route registration`
> 
> and add in description: `fix for issue #147` - it will link your PR with this issue
> 
> p.s thx for your help.

@etroynov should be done, I've also improved on the verb specific test by running it against all methods exported in the methods module used by the router.

**Review by etroynov — APPROVED** [review]

**Review by SagnikPradhan — APPROVED** [review]
You could probably move those checks to a function `isValidPath` instead but otherwise looks alright. 👍🏻

**Comment by peterver** [comment]
> You could probably move those checks to a function `isValidPath` instead but otherwise looks alright. 👍🏻

@SagnikPradhan Agree :). I'll take a look either tonight or tomorrow to move those bits into utility functions, there's some other parts of the validation which could use some abstraction to ensure code readability, of course I'll write some tests for any PR I open, thanks for the tip :).

**Pull request merged**